homeassistant/components/mqtt/device_trigger.py

```diff
@@ -66,23 +66,53 @@
 )
 
 DEVICE_TRIGGERS = "mqtt_device_triggers"
-ATTACHED_DEVICE_TRIGGERS = "mqtt_attached_device_triggers"
+
+
+@attr.s(slots=True)
+class AttachedTrigger:
+    """Attached trigger settings."""
+
+    action = attr.ib(type=AutomationActionType)
+    automation_info = attr.ib(type=dict)
+    remove = attr.ib(type=CALLBACK_TYPE)
 
 
 @attr.s(slots=True)
 class Trigger:
     """Device trigger settings."""
 
+    attached_triggers = attr.ib(type=[AttachedTrigger])
     device_id = attr.ib(type=str)
-    discovery_id = attr.ib(type=str)
     hass = attr.ib(type=HomeAssistantType)
     payload = attr.ib(type=str)
     qos = attr.ib(type=int)
     subtype = attr.ib(type=str)
     topic = attr.ib(type=str)
     type = attr.ib(type=str)
 
-    async def attach_trigger(self, action, automation_info):
+    async def add_trigger(self, action, automation_info):
```

Comment:
The function is called `add_trigger`, but all the variable names are talking about attaching a trigger. Should this be called `async_attach_trigger` ? 

And is it `automation_info` or `automation_config` ?

Reply:
It should be `automation_info`